### ISSUE TITLE
Plone 5 - installing the "Archetypes-based default content for Plone" via the advanced create site option, fails

### ISSUE THREAD

**Issue by witekdev** [plone/Products.CMFPlone#1519]
On a clean Plone 5.0.4 (from the unified installer), I am creating a new site using the advanced option. If **Archetypes-based default content for Plone** add-on is checked, I get this error:

2016-04-08 22:51:33 ERROR Zope.SiteErrorLog 1460173893.210.876095372669 http://localhost:8504/@@plone-addsite
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module Products.CMFPlone.browser.admin, line 257, in __call__
  Module Products.CMFPlone.factory, line 114, in addPloneSite
  Module Products.GenericSetup.tool, line 379, in runAllImportStepsFromProfile
   - __traceback_info__: profile-Products.ATContentTypes:content
  Module Products.GenericSetup.tool, line 1414, in _runImportStepsFromContext
  Module Products.GenericSetup.tool, line 1226, in _doRunImportStep
   - __traceback_info__: atct_content
  Module Products.ATContentTypes.setuphandlers, line 284, in importContent
  Module Products.ATContentTypes.setuphandlers, line 259, in setupPortalContent
AttributeError: unmarkCreationFlag

**Comment by khanstark** [comment]
@witekdev thanks for reporting this.
I think the codebase needs further review. Meanwhile u can try this:
<ul><li>Goto your installed plone directory and search for <b>setuphandlers.py</b> file</li>
<li>Open it up and browse to line 259 (<i>i.e. members.unmarkCreationFlag()</i>) and delete it!</li>
<li>Re-run your server, create new site again.</li>
</ul>
this might get things up and running but more digging is required.

**Comment by witekdev** [comment]
@khanstark Thanks, that work around works.
 
I will just add here (and perhaps this needs to be re-filed as a separate issue?)
When I add either of these add-on products using the advanced create a site option:
**Archetypes Content Types for Plone
Archetypes-based default content for Plone **

I see this visual glitch in the add new menu:

![screenshot from 2016-04-09 18 22 52](https://cloud.githubusercontent.com/assets/883619/14407069/5de4dd82-fe81-11e5-865a-c531e721cb0b.png)

**Comment by khanstark** [comment]
@witekdev even on creating a site without addons, `Collection` link seems to be messed up.
It needs to be rectified.

**Comment by khanstark** [comment]
See issue #1536

**Comment by witekdev** [comment]
@khanstark 
before closing this issue, did you verify that they two are the same issue?
In my opinion (but I haven't tested thoroughly), https://github.com/plone/Products.CMFPlone/issues/1536 is a misalignment of the collection label, this issue displays collection twice (which only happens when installing these add-on products specified in this ticket. I'm not convinced the two are same issue, but are indeed similar, and of course both issues are interacting with one another, so we see the misalignment of the collection label (but only once) here too.

**Comment by mauritsvanrees** [comment]
There are now three issues in this single issue... :-)

The original issue was a traceback when installing Archetypes-based default content for Plone in the advanced Plone add-site form.
I have fixed this in https://github.com/plone/Products.ATContentTypes/commit/4b17647d7365275022cbb3858204f0df19f6e0d9
(Yes, this was a direct commit on master, but for such a small change I felt lucky.)

**Comment by khanstark** [comment]
@witekdev I see the main motive of this thread was to remove errors that rose when creating a site with addons, which imo was already resolved!
The latter part of this thread comprises something else entirely, which infact would not be a bigger problem to solve once #1536 is fixed.

**Comment by witekdev** [comment]
@khanstark @mauritsvanrees you are both indeed correct. ok, let's see what issues remain once https://github.com/plone/Products.CMFPlone/issues/1536 is resolved.
